packages/block-editor/src/components/block-navigation/list.js

```diff
@@ -10,12 +10,35 @@ import classnames from 'classnames';
 import { Button } from '@wordpress/components';
 import { getBlockType } from '@wordpress/blocks';
 import { __ } from '@wordpress/i18n';
+import { create, getTextContent } from '@wordpress/rich-text';
 
 /**
  * Internal dependencies
  */
 import BlockIcon from '../block-icon';
 
+/**
+ * Get the block display name, if it has one, or the block title if it doesn't.
+ *
+ * @param {Object} blockType  The block type.
+ * @param {Object} attributes The values of the block's attributes
+ *
+ * @return {string} The display name value.
+ */
+function getBlockDisplayName( blockType, attributes ) {
```

Comment:
I wonder if this should also be used as the aria label of the block container.

Reply:
Yep. One of things I noticed when adding the block movers is that they only use the block title to describe the block, which is inconsistent. I'll look at proposing something in a new PR.

Reply:
@youknowriad #18132 attempts that.